GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -699,20 +695,99 @@ public class Result
             public const int SuccessCode = 0;
             public const int GenericFailureCode = 1;
 
-            public Result(string output, string errors, int returnCode)
+            public Result(string stdout, string stderr, int exitCode)
             {
-                this.Output = output;
-                this.Errors = errors;
-                this.ReturnCode = returnCode;
+                this.Output = stdout;
+                this.Errors = stderr;
+                this.ExitCode = exitCode;
             }
 
             public string Output { get; }
             public string Errors { get; }
-            public int ReturnCode { get; }
+            public int ExitCode { get; }
+
+            public bool ExitCodeIsSuccess
+            {
+                get { return this.ExitCode == Result.SuccessCode; }
+            }
+
+            public bool ExitCodeIsFailure
+            {
+                get { return !this.ExitCodeIsSuccess; }
+            }
+
+            public bool StderrContainsErrors()
+            {
+                if (!string.IsNullOrWhiteSpace(this.Errors))
+                {
+                    return !this.Errors
+                        .Split(new char[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries)
+                        .All(line => line.TrimStart().StartsWith("warning:", StringComparison.OrdinalIgnoreCase));
+                }
+
+                return false;
+            }
+        }
+
+        public class ConfigResult
+        {
+            private readonly Result result;
+            private readonly string configName;
+
+            public ConfigResult(Result result, string configName)
+            {
+                this.result = result;
+                this.configName = configName;
+            }
+
+            public bool TryParseAsString(out string value, out string error)
+            {
+                value = null;
+                error = string.Empty;
+
+                if (this.result.ExitCodeIsFailure && this.result.StderrContainsErrors())
+                {
+                    error = "Error while reading '" + this.configName + "' from config: " + this.result.Errors;
+                    return false;
+                }
 
-            public bool HasErrors
+                if (this.result.ExitCodeIsSuccess)
+                {
+                    value = this.result.Output;
+                }
+

```

Comment:
Should `TryParseAsString` have a `defaultValue` parameter like `TryParseAsInt`?

Reply:
Yes, good thought. Making that change helped clarify other bits that didn't feel so great. :)